java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -1558,15 +1563,43 @@ public CompletableFuture<Long> pexpiretime(@NonNull GlideString key) {
 
     @Override
     public CompletableFuture<Object> invokeScript(@NonNull Script script) {
-        return commandManager.submitScript(
-                script, List.of(), List.of(), this::handleObjectOrNullResponse);
+        if (script.getBinarySafeOutput()) {
+            return commandManager.submitScript(
+                    script, List.of(), List.of(), this::handleObjectOrNullResponse);
+        } else {
+            return commandManager.submitScript(
+                    script, List.of(), List.of(), this::handleBinaryObjectOrNullResponse);
```

Comment:
`this::handleBinaryObjectOrNullResponse` should be used in the  `getBinarySafeOutput()` branch